File: 2015/src/day2.py

def partOne(dimensions):
    sum = 0
    for l, w, h in dimensions:
        l, w, h = int(l), int(w), int(h)
        sum += 2*l*w + 2*w*h + 2*h*l + min(l*w, w*h, h*l)
    return sum


def partTwo(dimensions):
    sum = 0
    for l, w, h in dimensions:
        l, w, h = int(l), int(w), int(h)
        x = [l, w, h]
        x.sort()
        sum += (2*x[0]+2*x[1]) + (l*w*h)
    return sum


if __name__ == "__main__":
    with open("./input/day2.txt") as f:
        dimensions = f.readlines()
        dimensions = [x.rstrip().split('x') for x in dimensions]
    print(partOne(dimensions))
    print(partTwo(dimensions))

File: 2017/app/src/main/kotlin/aoc/solutions/Day2.kt
package aoc.solutions

import aoc.utils.FileLoader

fun solveDayTwo() {
    val input = FileLoader("day2.txt").readFile()

    println("Day 2 part 1: ${partOne(input)}")
    println("Day 2 part 2: ${partTwo(input)}")
}

private fun partOne(input: String): Int {
    var checkSum = 0

    for (line in input.split("\n")) {
        val nums = line.split("\\s+".toRegex()).map { it.toInt() }
        checkSum += (nums.max() - nums.min())
    }
    return checkSum
}

private fun partTwo(input: String): Int {
    var checkSum = 0
    for (line in input.split("\n")) {
        val nums = line.split("\\s+".toRegex()).map { it.toInt() }
        for (i in nums.indices) {
            for (j in nums.indices) {
                if (i == j) {
                    continue
                }
                if (nums[i] % nums[j] == 0) {
                    checkSum += nums[i] / nums[j]
                }
            }
        }
    }

    return checkSum
}
